src/uu/sort/src/sort.rs

```diff
@@ -454,72 +534,180 @@ fn compare_by(a: &str, b: &str, settings: &Settings) -> Ordering {
         (a, b)
     };
 
-    for compare_fn in &settings.compare_fns {
-        let cmp: Ordering = if settings.random {
-            random_shuffle(a, b, settings.salt.clone())
+    // 1st Compare
+    let mut cmp: Ordering = if settings.random {
+        random_shuffle(a, b, settings.salt.clone())
+    } else {
+        (settings.compare_fn)(a, b)
+    };
+
+    // Call "last resort compare" on any equal
+    if cmp == Ordering::Equal {
+        if settings.random || settings.stable || settings.unique {
+            cmp = Ordering::Equal
         } else {
-            compare_fn(a, b)
+            cmp = default_compare(a, b)
         };
-        if cmp != Ordering::Equal {
-            if settings.reverse {
-                return cmp.reverse();
-            } else {
-                return cmp;
-            }
-        }
+    };
+
+    if settings.reverse {
+        return cmp.reverse();
+    } else {
+        return cmp;
     }
-    Ordering::Equal
 }
 
+// Test output against BSDs and GNU with env var
+// lc_ctype=utf-8 until locales are implemented
+#[inline(always)]
 fn default_compare(a: &str, b: &str) -> Ordering {
     a.cmp(b)
 }
 
-fn get_leading_number(a: &str) -> &str {
+#[inline(always)]
+fn leading_num_common(a: &str) -> &str {
     let mut s = "";
-    for c in a.chars() {
-        if !c.is_numeric() && !c.eq(&'-') && !c.eq(&' ') && !c.eq(&'.') && !c.eq(&',') {
-            s = a.trim().split(c).next().unwrap();
+    // Strip string of non-numeric trailing chars
+    for (idx, c) in a.char_indices() {
+        if !c.is_numeric()
+            && !c.is_whitespace()
+            && !c.eq(&DECIMAL_PT)
+            && !c.eq(&THOUSANDS_SEP)
+            // check for e notation
+            && !c.eq(&'e')
+            && !c.eq(&'E')
+            && !a.chars().nth(0).unwrap_or('\0').eq(&POSITIVE)
+            && !a.chars().nth(0).unwrap_or('\0').eq(&NEGATIVE)
+        {
+            s = &a[..idx];
             break;
         }
-        s = a.trim();
+        s = a;
     }
-    return s;
+    s
 }
 
-// Matches GNU behavior, see:
-// https://www.gnu.org/software/coreutils/manual/html_node/sort-invocation.html
-// Specifically *not* the same as sort -n | uniq
-fn num_sort_dedup(a: &str) -> &str {
-    // Empty lines are dumped
-    if a.is_empty() {
-        return "0";
-    // And lines that don't begin numerically are dumped
-    } else if !a.trim().chars().nth(0).unwrap_or('\0').is_numeric() {
-        return "0";
+#[inline(always)]
+fn get_leading_num(a: &str) -> &str {
```

Comment:
can you document the function?
see
https://doc.rust-lang.org/rustdoc/what-is-rustdoc.html

Reply:
Pleased to do so.